### PR TITLE
Fix footer-small bug w.r.t. dynamic updates of children

### LINKED ISSUE #1549: Footer-small react - when rerender, new disclaimer-item content is ignored
Using v **@axa-ch/footer-small 3.0.8**

Under react: when switching language, and thus providing `AXAFooterSmall` with updated attributes:
- `<span slot="copyright">` is correctly updated (see italian content in screenshot)
- `<a slot="disclaimer-item">` isn't updated -- displays previous content (see english content in screenshot)

**Screenshot**

image

**Code**

```
<AXAFooterSmall
        onLanguageClick={handleLanguageClick}
        onDisclaimerClick={handleLegalClick}
        dynamic
      >
        {languageOptions.map(l => (
          <a
            key={l.name}
            slot="language-item"
            className={l.isActive ? 'm-footer-small__link--active' : undefined}
            href="#"
          >
            {l.name.toUpperCase()}
          </a>
        ))}
        {legals.map(l => (
          <a
            key={l.name}
            slot="disclaimer-item"
            href={l.url}
            target="_blank"
          >
            {l.name}
          </a>
        ))}
        <span slot="copyright">
          {copyrightText}
        </span>
      </AXAFooterSmall>
```

### PR DESCRIPTION
**Pull request by markus-walther**
Fixes #1549.

Needs IE 11 test.

### CODE REVIEW

**Review by LucaMele — DISMISSED** [review]
Tested on IE: It works as expected.

Solutions looks very nice! Ready to merge for me

File: src/components/20-molecules/footer-small/index.js [hunk 1]
```diff
@@ -43,45 +44,34 @@ class AXAFooterSmall extends InlineStyles {
     if (this.dynamic) {
       ev.preventDefault();
       this.onLanguageClick(languageIndex);
-      this.shadowRoot
```
Comment (raphaellueckl) [inline src/components/20-molecules/footer-small/index.js:46]: I guess the removal of this block is the reason, why the active-language indicator does not update anymore.

To be seen here: https://axa-ch.github.io/plib-feature/bugfix/footer-small-dynamic-children/?path=/story/molecules-footer-small-react--footer-with-callbacks-on-language

And here: https://axa-ch.github.io/plib-feature/bugfix/footer-small-dynamic-children/?path=/story/molecules-footer-small-demos--footer-small-dynamic-links

I checked with the current prod and there it works. Interestingly, this dynamic example from your branch seems to work: 
https://axa-ch.github.io/plib-feature/bugfix/footer-small-dynamic-children/?path=/story/molecules-footer-small-react--footer-with-dynamic-change-of-children-upon-language-change

**Review by raphaellueckl — CHANGES_REQUESTED** [review]
A very nice PR Markus! And a nice approach indeed. I just found a small issue that needs a bit of attention, but thank you for solving this problem! It was giving me headaches.

File: src/components/20-molecules/footer-small/index.js (hunk 1 above)
Comment (markus-walther) [reply src/components/20-molecules/footer-small/index.js:46]: Fixed with new commit, thanks!

Comment (markus-walther) [reply src/components/20-molecules/footer-small/index.js:46]: Again fixed with new commit, sorry...

**Review by raphaellueckl — APPROVED** [review]

**Pull request merged**